socorro/external/es/crashstorage.py

```diff
@@ -138,22 +143,91 @@ def _submit_crash_to_elasticsearch(self, connection, crash_document):
             index_creator.create_socorro_index(es_index)
 
         # Submit the crash for indexing.
-        try:
-            connection.index(
-                index=es_index,
-                doc_type=es_doctype,
-                body=crash_document,
-                id=crash_id
-            )
+        success = False
+        # Don't retry more than 5 times. That is to avoid infinite loops in
+        # case of an unhandled exception.
+        retries_pool = [True] * 5 + [False]
+        while not success and retries_pool.pop(0):
+            try:
+                connection.index(
+                    index=es_index,
+                    doc_type=es_doctype,
+                    body=crash_document,
+                    id=crash_id
+                )
+                success = True
+            except elasticsearch.exceptions.TransportError as e:
+                field_name = None
+
+                if 'MaxBytesLengthExceededException' in e.error:
+                    # This is caused by a string that is way too long for
+                    # Elasticsearch.
+                    matches = re.findall(
+                        FIELD_NAME_STRING_ERROR_REGEX, e.error
```

Comment:
Wow! I didn't even know this was possible!
I.e. 
```python
>>> re.findall('\d', 'n00dle')
['0', '0']
>>> re.findall(re.compile('\d'), 'n00dle')
['0', '0']
```

But you don't need to double-wrap it anyway. 
Just use `matches = FIELD_NAME_STRING_ERROR_REGEX.findall(e.error)`